src/network/MissionControlProtocol.cpp

```diff
@@ -221,6 +224,25 @@ void MissionControlProtocol::sendRoverPos() {
 	}
 }
 
+static bool validateWaypointNavRequest(const json& j) {
+	return util::validateKey(j, "latitude", val_t::number_unsigned) &&
+		   util::validateKey(j, "longitude", val_t::number_unsigned) &&
+		   util::validateKey(j, "approximate", val_t::boolean) &&
+		   util::validateKey(j, "gated", val_t::boolean);
```

Comment:
According to the MC readme, these fields should be called `isGate` and `isApproximate`, is that not the case?

Reply:
This was just a quick fix. They are not called isGate and isApproximate in the mission control code. They got changed somewhere along the line. The current message uses gate and approximate. Either the MC readme needs to change or the JS code on MC side